packages/core/core/src/AssetGraph.js

```diff
@@ -163,9 +172,32 @@ export default class AssetGraph extends Graph<AssetGraphNode> {
     let requestNode = nodeFromTransformerRequest(req);
     let {added, removed} = this.replaceNodesConnectedTo(depNode, [requestNode]);
 
+    // Defer transforming this dependency if it is marked as weak, there are no side effects,
+    // and no re-exported symbols are used by ancestor dependencies.
+    // This helps with performance building large libraries like `lodash-es`, which re-exports
+    // a huge number of functions since we can avoid even transforming the files that aren't used.
+    let defer = false;
+    if (dep.isWeak && req.sideEffects === false) {
```

Comment:
This code is confusing and probably a little specific to go in the AssetGraph itself. Would like to come up with a better abstraction for storing symbols in the graph.